tests/time-handling/SimpleMaxTime.cpp

```diff
@@ -0,0 +1,45 @@
+#ifndef PRECICE_NO_MPI
+
+#include "testing/Testing.hpp"
+
+#include <precice/precice.hpp>
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(TimeHandling)
+BOOST_AUTO_TEST_CASE(SimpleMaxTime, *boost::unit_test::tolerance(1e-18))
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank));
+
+  precice::Participant p(context.name, context.config(), context.rank, context.size);
+  double               coord[] = {0, 0};
+  p.setMeshVertex(context.name + "-Mesh", coord);
+  p.initialize();
+
+  const double expected = 0.01;
+
+  BOOST_TEST(p.getMaxTimeStepSize() == expected);
+  BOOST_TEST(p.isCouplingOngoing());
+  p.advance(p.getMaxTimeStepSize());
+
+  BOOST_TEST(p.getMaxTimeStepSize() == expected);
+  BOOST_TEST(p.isCouplingOngoing());
+  p.advance(p.getMaxTimeStepSize());
+
+  BOOST_REQUIRE(p.isCouplingOngoing());
+  // BUG was: return value 0.009999999999999998
+  BOOST_TEST(p.getMaxTimeStepSize() == expected);
+  p.advance(p.getMaxTimeStepSize());
```

Comment:
I think whether this high accuracy is an overkill or not depends a lot on what we expect from preCICE here. If I tell preCICE that `<time-window-size value="0.01" />`, is there a good reason to accept preCICE returning something else than `0.01`?